### ISSUE TITLE
Allow to set custom sort on Jet Smart Filters

### ISSUE THREAD

**Issue by dorshiff** [CrocoBlock/suggestions#2312]
When adding multiple filters to a single Checkboxes Filter element, there is no way to set their order.
The filters appear in the frontend in random order.

![image](https://user-images.githubusercontent.com/56870532/102049238-2d342b00-3de9-11eb-9f30-a3fb6700db87.png)

**Comment by danielemarucci** [comment]
Hi, I also have the need to sort filter entries. Is there any news on this or possibility to sort in other way (e.g. with custom field or by number of results)?

example:
![IMAGE 2021-10-06 11:29:17](https://user-images.githubusercontent.com/53402499/136177172-1b038900-0afd-4a3b-adec-ef016dfda8d9.jpg)

**Comment by stasonua0** [comment]
A very demanded function, please increase its execution priority

**Comment by kskonovalov** [comment]
+1 for this thing

**Comment by philc100** [comment]
I've needed this on the select filter forever. Filtering a huge list of products and they are displayed in date order. Horrible for end user.

**Comment by mennybe** [comment]
+1 for this thing

I'll be more than happy to have the ability of choosing the field order

**Comment by antoinelbch** [comment]
+1

**Comment by kevanbulmer** [comment]
+1

**Comment by tealou** [comment]
+1 - any progress? This is something that we've been waiting years for with no solution and it is a pretty basic thing. It is these details that make me want to move away from Crocoblock and wasting days on end for basic functionality is wearing thin.

**Comment by w3b-beweb** [comment]
+1 for this one. I need to order a checkboxes categories filter by term_order, not alphabetically

**Comment by bombarda-rpd** [comment]
+1 for this. I need to order a checkboxes categories filter by term_order, not alphabetically

**Comment by Clauded2** [comment]
+1 or adding an 'external provider' that works with query ID of the one created with the QueryBuilder.

**Comment by mcpixl** [comment]
+1 - For everyone wondering when this will become a feature, the answer is never. Crocoblock has now taken on too many plugins which it cannot maintain causing many basic crucial features to be missing. The only way to fix this currently is to take the plugin and make it your own however, this is preventing those without coding experience from utilising it. I hope this simple feature gets added but I doubt it ever will.

**Comment by stasonua0** [comment]
It seems that all forces are thrown on a jet engine, and other analogies are forgotten.

**Comment by labachev** [comment]
Since this is not a priority for the developers I can suggest you a workaround for that:

In the name of the taxonomy you use, you can add a number and a special symbol in the beginning.
After that, you can put something like this in functions.php

```
<php
function my_jet_filter_options( $options, $filter_id, $filter ){

    if($filter_id == 123){
        asort( $options );
    }
	$options = str_replace('1.', '', $options);
    return $options;

}
add_filter( 'jet-smart-filters/filters/filter-options', 'my_jet_filter_options',10,3 );

```

I hope this helps.

**Comment by baguet** [comment]
Thank you so much for your help. This trick saved me hours.

**Comment by aftergroup** [comment]
**[Custom Taxonomy Order by Marcel Pol](https://wordpress.org/plugins/custom-taxonomy-order-ne/)**

The plugin enables you to **easily rearrange taxonomies** by employing a straightforward drag-and-drop functionality. After you finish reordering them, the taxonomy terms will be displayed in your JetSmartFilters filter according to the precise order you have specified.

Cheers!

**Comment by ismdibrahim** [comment]
You can try this. It's working for me. 

function my_jet_filter_options( $options, $filter_id, $filter ){
	
    if($filter_id == 29320 || $filter_id == 29221 || $filter_id == 29321){
		$sorted_category = wp_list_sort( $options, 'name', 'ASC' );
		return $sorted_category;
    }
}
add_filter( 'jet-smart-filters/filters/filter-options', 'my_jet_filter_options',10,3 );

**Comment by dani88dani** [comment]
You can use the "Custom Taxonomy Order" plugin as posted above.

It worked perfectly for me to customize the order of terms (Product Categories and Attributes) used in Checkbox List filters. Just sort the items manually and pick "custom order" from the settings below and this will work instantly.

**Comment by zehjotkah** [comment]
If you don't want to use additional plugins or code, you can set up a query via JetEngine Query Builder where you define the order and then use this query as the data source of the filter. 

For a completely custom order you can use these settings in the query: 
under general: Order By: "Match the order of the 'Slug' param" or "MAtch the order of the 'Include' param"
under Include/Exclude: type in the Slugs or IDs in the preferred order.

Then you can use this query as the data source of the Smart Filter like described in this article: https://crocoblock.com/knowledge-base/jetengine/how-to-create-filters-with-query-builder-source/

**Comment by WebblaJordan** [comment]
> If you don't want to use additional plugins or code, you can set up a query via JetEngine Query Builder where you define the order and then use this query as the data source of the filter.
> 
> For a completely custom order you can use these settings in the query: under general: Order By: "Match the order of the 'Slug' param" or "MAtch the order of the 'Include' param" under Include/Exclude: type in the Slugs or IDs in the preferred order.
> 
> Then you can use this query as the data source of the Smart Filter like described in this article: https://crocoblock.com/knowledge-base/jetengine/how-to-create-filters-with-query-builder-source/

This worked perfectly, thanks!

**Comment by Crocoblock** [comment]
In the next update, we will add options for filter order. This will be available for checkbox, select, and radio filters https://prnt.sc/iYsGXI9FnjbU https://prnt.sc/fdl_2dVC7ANM As for the order of multifilter display, we can't implement it right now because there are dependencies on the elementor plugin

**Comment by bombarda-rpd** [comment]
Super! Thanks!

**Comment by Crocoblock** [comment]
added and released in jetSmartFilters 3.6.0
